code/_globalvars/lists/ai_laws.dm

```diff
@@ -0,0 +1,21 @@
+GLOBAL_LIST_INIT(lawset_datums, get_lawset_datums())
+
+/proc/get_lawset_datums()
+	. = list()
+
+	for(var/lawset in subtypesof(/datum/ai_laws))
+		.[lawset] = new lawset
+
+GLOBAL_LIST_INIT(lawset_law_lists, get_lawset_law_lists())
+
+/proc/get_lawset_law_lists()
+	. = list()
+
+	var/datum/ai_laws/lawset_datum = new /datum/ai_laws
+
+	for(var/lawset in subtypesof(/datum/ai_laws))
+		lawset_datum.set_laws_lawset(lawset)
+
+		.[lawset_datum.name] = lawset_datum.get_law_list(TRUE, FALSE, FALSE).Copy()
```

Comment:
I prefer this syntax, do you want me to adjust for the linter?
`warning: proc call requires static type: "Copy"`

Reply:
- To satisfy the linter that this `.Copy` call is valid, add this line to the top of `get_law_list`:
  `RETURN_TYPE(/list)`

- However, the `.Copy` call here is unnecessary, as `get_law_list` always returns a brand-new list.

- To satisfy the linter about the other `.Copy` calls,
  change `GLOBAL_LIST_INIT(lawset_datums, get_lawset_datums())`
  to `GLOBAL_LIST_INIT_TYPED(lawset_datums, /datum/ai_laws, get_lawset_datums())`